### PR TITLE
Ready to grade

### PR DESCRIPTION
**Pull request by PatriceBriggs**
HI James,
Ready to grade - I don't think the delete works right - think it deletes everything...

Patrice

### CODE REVIEW

File: routes/truckRoutes.js [hunk 1]
```diff
@@ -1,3 +1,74 @@
...
+
+    .put(function (request, response) {
+        var truckId = request.params.id;
+        trucks.updateTruck(truckName);
+        response.sendstatus(200);
+    });
+
+
+
+module.export = router;
```
Comment (jamdon) [inline routes/truckRoutes.js]: You're missing an 's':

`module.exports = router;`

It's not exporting the router middle-ware properly,  causing the error you're seeing.

File: routes/truckRoutes.js [hunk 2]
```diff
@@ -1,3 +1,74 @@
...
+        	}
+        });
+
+    })
+
+    .post( function (request, response){
+        var newTruck = request.body;
+        if (newTruck) {
+            newTruck.read = false;
+            trucks.addTruck(newTruck);
```
Comment (jamdon) [inline routes/truckRoutes.js]: You're not using the trucks module anymore, so this line won't work.

File: routes/truckRoutes.js [hunk 3]
```diff
@@ -1,3 +1,74 @@
...
+    .get(function (request, response) {         
+        Truck.findById(request.params.id, function (error, truck) {
+            if (error) {
+        		response.status(500).send(error);
+        	} else {
+        		response.json(truck);
+        	}
+        });   
+
+        var truck = trucks.getTruck(request.params.id);
```
Comment (jamdon) [inline routes/truckRoutes.js]: You're not using the trucks module anymore.

File: routes/truckRoutes.js [hunk 4]
```diff
@@ -1,3 +1,74 @@
...
+    	    response.status(404).json('Sorry, cannot find the truck called ' +
+    		    request.params.name);
+        } else {
+    	    response.json(truck);
+        }
+    })
+
+    .delete( function (request, response) {
+        var truckName = request.params.id;
+        trucks.removeTruck(truckName);
```
Comment (jamdon) [inline routes/truckRoutes.js]: Same as above.

File: routes/truckRoutes.js [hunk 5]
```diff
@@ -1,3 +1,74 @@
...
+        response.sendStatus(200);
+    })
+
+    .put(function (request, response) {
+        var truckId = request.params.id;
+        trucks.updateTruck(truckName);
```
Comment (jamdon) [inline routes/truckRoutes.js]: Same as above.

**Comment by jamdon** [comment]
The main issue you're having has to do with doing `module.export` instead of `module.exports` - I added comments about not using the trucks module as well, but I'm guessing you're in the middle of the homework and simply haven't gotten to that part yet.

**Comment by PatriceBriggs** [comment]
Updated all the routes to use mongoose. Error message that I'm trying to open an unclosed connection.  Not sure what that means.
Patrice

**Comment by jamdon** [comment]
That error is indicating that you're trying to connect without closing your current connection. You have a connection call in both app.js and truckRoutes.js. You only want to connect once unless you've explicitly disconnected at some point.

File: routes/truckRoutes.js [hunk 6]
```diff
@@ -1,3 +1,98 @@
...
+        });   
+    })
+
+    .delete(function(request, response) {
+
+        Truck.findById(request.params.id, function (error, truck) {
+            if (error) {
+                response.status(500).send(error);
+            } else {
+                Truck.remove(function (error) {
```
Comment (jamdon) [inline routes/truckRoutes.js:59]: when you call the findById method on the Truck module, you get the truck instance back as the second parameter to the callback - "truck". This is the instance you want to call remove on.

File: routes/truckRoutes.js [hunk 7]
```diff
@@ -1,3 +1,98 @@
...
+    })
+
+    .put(function(request, response) {
+
+    // like the GET route, use the findById method on the mongoose model     
+        Truck.findById(request.params.id, function (error, truck) {         
+            if (error) {            
+                 response.status(500).send(error);         
+            } else {             
+                Truck.name = request.body.name;
```
Comment (jamdon) [inline routes/truckRoutes.js:77]: Same deal here - you want to modify the "truck" instance that was passed as the second parameter to the callback, not the "Truck" module itself. When you're done, you call save on that truck instance, and send it back to the client.

**Comment by jamdon** [comment]
:+1:

**Pull request closed without merging**